### PR TITLE
Sword swing sound effect added

### PR DESCRIPTION
**Pull request by jakkeez**
Sword swing sound effect added when player presses left click. Test by running the Jacob test scene and pressing left click will play a *swing" sound.

### CODE REVIEW

**Review by jocke1995 — DISMISSED** [review]
works

File: StortSpelProjekt/Sandbox/src/main.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE, LPSTR, int nCmdShow)
 
     UpdateScene = &DefaultUpdateScene;
 
-    //Scene* jacobScene = JacobsTestScene(sceneManager);
+    Scene* jacobScene = JacobsTestScene(sceneManager);
```
Comment (Schtekt) [inline StortSpelProjekt/Sandbox/src/main.cpp:59]: Make sure to comment out "fredrikScene" or your own before merge.

File: StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.cpp [hunk 2]
```diff
@@ -23,6 +24,10 @@ component::MeleeComponent::MeleeComponent(Entity* parent) : Component(parent)
 	// Fetch the player transform
 	m_pMeleeTransform = parent->GetComponent<component::TransformComponent>()->GetTransform();
 
+	// Fetch the player audio component
+	m_pVoiceComponent = parent->GetComponent<component::Audio2DVoiceComponent>();
+	m_pVoiceComponent->AddVoice(L"SwordSwing");
+
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.cpp]: Assuming that there will be a swordswing makes other scenes crash.
What would happen if an enemy has a meleecomponent and does not use this sound?

File: StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.cpp [hunk 3]
```diff
@@ -45,17 +50,17 @@ void component::MeleeComponent::OnUnInitScene()
 void component::MeleeComponent::Update(double dt)
 {
 	// Takes the transform of the player cube and moves it forward to act as a hitbox
-	m_MeleeTransformTwo = *m_pMeleeTransform;
+	m_MeleeTransformModified = *m_pMeleeTransform;
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.cpp:62]: Love the namechange!

File: StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.cpp [hunk 4]
```diff
@@ -76,15 +81,17 @@ void component::MeleeComponent::Update(double dt)
...
 	m_Hitbox = temp;
 
 }
 
 void component::MeleeComponent::Attack(bool attack)
 {
+	
 	if (!m_Cooldown)
 	{
+		m_pVoiceComponent->Play(L"SwordSwing");
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.cpp]: Again, assuming that the component has this sound.

**Review by Schtekt — CHANGES_REQUESTED** [review]
The idea itself is good, but needs some rethinking on how it is executed. Meleecomponent should not assume that it has an audiocomponent from it's parent.

**Review by Schtekt — APPROVED** [review]

**Review by jocke1995 — APPROVED** [review]

**Pull request merged**